crates/sui-core/src/authority_aggregator.rs

```diff
@@ -1056,14 +1069,22 @@ where
                                     // we cannot retry.
                                     state.object_or_package_not_found_stake += weight;
                                 }
+                                else if err.is_overload() {
+                                    // Special case for validator overload too. Once we have >= 2f + 1
+                                    // overloaded validators we consider the system overloaded and we
+                                    // don't retry.
+                                    state.overloaded_stake += weight;
+                                }
                                 else if !retryable && !self.record_conflicting_transaction_if_any(&mut state, name, weight, &err) {
                                     // Error is neither retryable or a potentially retryable conflicting transaction.
                                     state.non_retryable_stake += weight;
                                 }
                                 state.errors.push((err, vec![name], weight));
 
-                                if state.non_retryable_stake >= validity_threshold || state.object_or_package_not_found_stake >= quorum_threshold {
-                                    // We have hit an exit condition, f+1 non-retryable err or 2f+1 object not found,
+                                if state.non_retryable_stake >= validity_threshold
+                                    || state.object_or_package_not_found_stake >= quorum_threshold
+                                    || state.overloaded_stake >= quorum_threshold {
```

Comment:
Wonder if you want to separate this check and only have it return `ReduceOutput::Failed(state)` without setting `state.retryable = false` just to make it clearer that we don't consider this case as non-retryable

Reply:
I'd say it's still non-retryable at QD level. It's just that we return something other than `FatalTransaction`.